src/main/java/org/springframework/data/util/TypeCollector.java

```diff
@@ -125,6 +127,10 @@ Set<Type> visitConstructorsOfType(ResolvableType type) {
 		}
 		Set<Type> discoveredTypes = new LinkedHashSet<>();
 		for (Constructor<?> constructor : type.toClass().getDeclaredConstructors()) {
+
+			if (constructor.isSynthetic()) {
```

Comment:
Kotlin likes to generate synthetic constructors for data classes. We should make sure to not break stuff. Likely, the field-based inspection is going to catch such cases.

Reply:
See: https://github.com/spring-projects/spring-data-commons/blob/6a23723f07669e5d4031b3378b3af40e0d15eb82/src/main/java/org/springframework/data/mapping/model/KotlinClassGeneratingEntityInstantiator.java#L108